compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageCompilation.java

```diff
@@ -210,4 +221,73 @@ private static void addCompilerPluginDiagnostics(PackageCompilation compilation,
+     */
+    private static void reportNonExportedModules(PackageCompilation compilation) {
+        ResolvedPackageDependency resolvedPackageDependency = new ResolvedPackageDependency(
+                compilation.rootPackageContext.project().currentPackage(), PackageDependencyScope.DEFAULT);
+        Collection<ResolvedPackageDependency> directDependencies = compilation.packageResolution.dependencyGraph()
+                .getDirectDependencies(resolvedPackageDependency);
+
+        List<Diagnostic> allDiagnostics = new ArrayList<>(compilation.diagnosticResult.diagnostics());
+        List<Diagnostic> nonExportedModulesDiagnostics = new ArrayList<>();
+
+        for (ModuleId moduleId: compilation.rootPackageContext.project().currentPackage().moduleIds()) {
+            Module module = compilation.rootPackageContext.project().currentPackage().module(moduleId);
+
+            if (DOT.equals(module.moduleName().toString())) {
+                continue;
+            }
+
+            for (DocumentId documentId : module.documentIds()) {
+                Document document = module.document(documentId);
+                ModulePartNode modulePartNode = document.syntaxTree().rootNode();
+
+                for (ImportDeclarationNode importDcl : modulePartNode.imports()) {
+                    if (importDcl.orgName().isPresent()) {
+                        // foo/winery.storage
+                        String orgName = importDcl.orgName().get().orgName().text(); // foo
+
+                        String pkgName;
+                        String moduleName;
+                        if (importDcl.moduleName().size() == 1) {
+                            pkgName = importDcl.moduleName().get(0).text(); // winery
+                            moduleName = pkgName; // winery
+                        } else if (importDcl.moduleName().size() == 2) {
+                            pkgName = importDcl.moduleName().get(0).text(); // winery
+                            moduleName = pkgName + "." + importDcl.moduleName().get(1).text(); // winery.storage
+                        } else {
+                            throw new ProjectException("invalid import declaration:" + importDcl.moduleName());
+                        }
+
+                        for (ResolvedPackageDependency pkgDependency : directDependencies) {
+                            if (orgName.equals(pkgDependency.packageInstance().descriptor().org().value())
+                                    && pkgName.equals(pkgDependency.packageInstance().descriptor().name().value())) {
+                                List<String> exportedModuleNames = pkgDependency.packageInstance().manifest().export();
+
+                                if (!exportedModuleNames.contains(moduleName)) {
+                                    Diagnostic diagnostic = DiagnosticFactory.createDiagnostic(new DiagnosticInfo(
+                                        null, "module '" + moduleName + "' is not an exported module",
```

Comment:
Don't use null here, Please use a proper error code. This error code is shown in VSCode diagnostics. We don't want to show `null` to users.